File: app/src/main/java/me/huar/jetpack_demo/data/network/schedulers/BaseSchedulerProvider.java
package me.huar.jetpack_demo.data.network.schedulers;

import io.reactivex.FlowableTransformer;
import io.reactivex.Scheduler;

public interface BaseSchedulerProvider {

    public Scheduler computation();

    public Scheduler io();

    public Scheduler ui();

    public <T> FlowableTransformer<T, T> applySchedulers();

}

File: app/src/main/java/me/huar/jetpack_demo/data/network/schedulers/BaseSchedulerProvider.kt
package me.huar.jetpack_demo.data.network.schedulers

import io.reactivex.FlowableTransformer
import io.reactivex.Scheduler

interface BaseSchedulerProvider {
    fun computation(): Scheduler?
    fun io(): Scheduler?
    fun ui(): Scheduler?
    fun <T> applySchedulers(): FlowableTransformer<T, T>?
}